src/main/java/dev/openfeature/javasdk/Structure.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    public Set<String> keySet() {
+        return this.attributes.keySet();
+    }
+
+    // getters
+    public Boolean getBooleanAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.BOOLEAN);
+    }
+
+    public String getStringAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.STRING);
+    }
+
+    public Integer getIntegerAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.INTEGER);
+    }
+
+    public Double getDoubleAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.DOUBLE);
+    }
+
+    public <T> List<T> getArrayAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.ARRAY);
+    }
+
+    public Structure getStructureAttribute(String key) {
+        return getAttributeByType(key, StructureFieldType.OBJECT);
+    }
+
+    /**
+     * Fetch date-time relevant key.
+     * 
+     * @param key feature key
+     * @return date time object.
+     * @throws java.time.format.DateTimeParseException if it's not a datetime
+     */
+    public ZonedDateTime getDatetimeAttribute(String key) {
+        String attr = getAttributeByType(key, StructureFieldType.STRING);
+        if (attr == null) {
+            return null;
+        }
+        return ZonedDateTime.parse(attr, DateTimeFormatter.ISO_ZONED_DATE_TIME);
+    }
+
+    // adders
+    public Structure add(String key, Boolean value) {
+        attributes.put(key, new Pair<>(StructureFieldType.BOOLEAN, value));
+        return this;
+    }
+
+    public Structure add(String key, String value) {
+        attributes.put(key, new Pair<>(StructureFieldType.STRING, value));
+        return this;
+    }
+
+    public Structure add(String key, Integer value) {
+        attributes.put(key, new Pair<>(StructureFieldType.INTEGER, value));
+        return this;
+    }
+
+    public Structure add(String key, Double value) {
+        attributes.put(key, new Pair<>(StructureFieldType.DOUBLE, value));
+        return this;
+    }
+
+    /** 
+     * Add date-time relevant key.
+     * 
+     * @param key feature key
+     * @param value date-time value
+     * @return Structure
+     */
+    public Structure add(String key, ZonedDateTime value) {
+        attributes.put(key, new Pair<>(StructureFieldType.STRING, value != null 
+            ? value.format(DateTimeFormatter.ISO_ZONED_DATE_TIME) : null));
+        return this;
+    }
+
+    public Structure add(String key, Structure value) {
+        attributes.put(key, new Pair<>(StructureFieldType.OBJECT, value));
+        return this;
+    }
+
+    public <T> Structure add(String key, List<T> value) {
+        attributes.put(key, new Pair<>(StructureFieldType.ARRAY, value));
+        return this;
+    }
+
+    /**
+     * Get all attributes, regardless of type.
+     * 
+     * @return all attributes on the structure
+     */
+    public Map<String, Object> getAllAttributes() {
```

Comment:
Would it be possible to get something like this that includes the type annotations? So that if one is converting this structure into something using another structure, or a builder, that they can iterate it and dispatch the methods to the correct types?

Reply:
@kinyoklion I think you should be able to do that with `instanceof`?

Reply:
> @kinyoklion I think you should be able to do that with instanceof?

Indeed, this is what I'm doing in my provider. Is there some reason it's better to leverage the underlying annotation in the `Pair`?

Reply:
Yeah, it can be done that way. Also the performance of `instanceof` seems to be good.

The information was already there, so having to immediately re-construct it, seemed a bit counter intuitive.

Reply:
Totally. I just don't want it to be part of a public API